test/transaction.test.ts

```diff
@@ -0,0 +1,138 @@
+import * as assert from 'assert';
+
+jest.setTimeout(50000);
+
+describe('buildTx', () => {
+  let senderUtxos: UtxoInterface[];
+  let USDT: string = '';
+
+  beforeAll(async () => {
+    await faucet(senderAddress);
+    // mint and fund with USDT
+    const minted = await mint(senderAddress, 100);
+    USDT = minted.asset;
+
+    await sleep(3000);
+
+    senderUtxos = await fetchAndUnblindUtxos(
+      [
+        {
+          address: senderAddress,
+          blindingKey: senderBlindingKey,
+        },
+      ],
+      APIURL
+    );
+  });
+
+  it('should build a confidential transaction spending USDT', () => {
+    // create a tx using wallet
+    const tx = senderWallet.createTx();
+
+    const outputs: OutputInterface[] = [
+      {
+        asset: USDT,
+        value: 50000,
+        script: address
+          .toOutputScript(recipientAddress, networks.regtest)
+          .toString('hex'),
+      },
+    ];
+
+    const unsignedTx = buildTx(
+      tx,
+      senderUtxos,
+      outputs,
+      // for change script, we just return the sender script for all assets
+      (_: string) =>
+        address.toOutputScript(senderAddress, networks.regtest).toString('hex')
+    );
+
+    assert.doesNotThrow(() => Psbt.fromBase64(unsignedTx));
+  });
+
+  it('should build a confidential transaction spending LBTC', async () => {
+    // create a tx using wallet
+    const tx = senderWallet.createTx();
+
+    const outputs: OutputInterface[] = [
+      {
+        asset: networks.regtest.assetHash,
+        value: 50000,
+        script: address
+          .toOutputScript(recipientAddress, networks.regtest)
+          .toString('hex'),
+      },
+    ];
+
+    const unsignedTx = buildTx(
+      tx,
+      senderUtxos,
+      outputs,
+      // for change script, we just return the sender script for all assets
+      (_: string) =>
+        address.toOutputScript(senderAddress, networks.regtest).toString('hex')
+    );
+
+    assert.doesNotThrow(() => Psbt.fromBase64(unsignedTx));
+  });
+
+  it('should be able to create a complex transaction and broadcast it', async () => {
+    const outputs = [
+      {
+        asset: networks.regtest.assetHash,
+        value: 50000,
+        script: address
+          .toOutputScript(recipientAddress, networks.regtest)
+          .toString('hex'),
+      },
+      {
+        asset: USDT,
+        value: 50000,
+        script: address
+          .toOutputScript(recipientAddress, networks.regtest)
+          .toString('hex'),
+      },
+    ];
+
+    const tx = senderWallet.createTx();
+    const unsignedTx = buildTx(
+      tx,
+      senderUtxos,
+      outputs,
+      // for change script, we just return the sender script for all assets
+      (_: string) =>
+        address.toOutputScript(senderAddress, networks.regtest).toString('hex'),
+      true
+    );
+
+    const pset = decodePset(unsignedTx);
+
+    const privKeyBuffer = Buffer.from(senderBlindingKey, 'hex');
```

Comment:
This looks like something the Identity should take care, like the Liquality Chain Abstraction Layer, a blindPset would be doable for both the Mnemonic and MasterPublicKey.

Will do in a separated PR.